lib/packages/core/lib/src/widgets/chatroom/chatroom_menu.dart

```diff
@@ -304,4 +316,87 @@ class _ChatroomMenuState extends State<LMChatroomMenu> {
       ),
     );
   }
+
+  void showBlockDialog(ChatroomAction action) {
+    showDialog(
+      context: context,
+      builder: (context) => LMChatDialog(
+        title: const Text("Block direct messaging"),
+        content: const Text(
+          'Are you sure you do not want to receive new messages from this user?',
+        ),
+        actions: [
+          LMChatText(
+            'Cancel',
+            onTap: () {
+              Navigator.pop(context);
+            },
+            style: const LMChatTextStyle(
+              maxLines: 1,
+              textStyle: TextStyle(
+                overflow: TextOverflow.ellipsis,
+                fontSize: 16,
+                color: Colors.black87,
+                fontWeight: FontWeight.w500,
+              ),
+            ),
+          ),
+          LMChatText(
+            'Confirm',
+            onTap: () {
+              blockDM(action);
+              Navigator.pop(context);
+            },
+            style: const LMChatTextStyle(
+              maxLines: 1,
+              textStyle: TextStyle(
+                overflow: TextOverflow.ellipsis,
+                fontSize: 16,
+                fontWeight: FontWeight.w500,
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  void blockDM(ChatroomAction action) async {
```

Comment:
we can move this logic to ChatroomActionBloc

Reply:
Will align it for later then, will be too big of a change now.

Reply:
currently the whole flow is little messy, it would be very hard to debug if anything occur in this module.

Reply:
It's only one function (this one). Everywhere else, there is nothing related to block member. 
And if we want to modify all of this we should ideally mute notifications using BLoC too. So, we will have to change all actions to follow that pattern.